src/propagate-tor-test_cuda.cu

```diff
@@ -0,0 +1,1048 @@
+   printf("track in cov: xx=%.2e, yy=%.2e, zz=%.2e \n", inputtrk.cov[SymOffsets66(PosInMtrx(0,0,6))],
+	                                       inputtrk.cov[SymOffsets66(PosInMtrx(1,1,6))],
+	                                       inputtrk.cov[SymOffsets66(PosInMtrx(2,2,6))]);
+   printf("hit in pos: x=%f, y=%f, z=%f, r=%f \n", inputhit.pos[0], inputhit.pos[1], inputhit.pos[2], sqrtf(inputhit.pos[0]*inputhit.pos[0] + inputhit.pos[1]*inputhit.pos[1]));
+   
+   printf("produce nevts=%i ntrks=%i smearing by=%f \n", nevts, ntrks, smear);
+   printf("NITER=%d\n", NITER);
+   long setup_start, setup_stop;
+   struct timeval timecheck;
+
+   gettimeofday(&timecheck, NULL);
+   setup_start = (long)timecheck.tv_sec * 1000 + (long)timecheck.tv_usec / 1000;
+   MPTRK* trk = prepareTracks(inputtrk);
+   MPHIT* hit = prepareHits(inputhit);
+   MPTRK* outtrk ;
+   cudaMallocHost((void**)&outtrk,nevts*nb*sizeof(MPTRK)); 
+   //device pointers
+   MPTRK* trk_dev;
+   MPHIT* hit_dev;
+   MPTRK* outtrk_dev;
+   cudaMalloc((MPTRK**)&trk_dev,nevts*nb*sizeof(MPTRK));
+   cudaMalloc((MPHIT**)&hit_dev,nlayer*nevts*nb*sizeof(MPHIT));
+   cudaMalloc((MPTRK**)&outtrk_dev,nevts*nb*sizeof(MPTRK));
+
+   //for (size_t ie=0;ie<nevts;++ie) {
+   //  for (size_t it=0;it<ntrks;++it) {
+   //    float x_ = x(trk,ie,it);
+   //    float y_ = y(trk,ie,it);
+   //    float z_ = z(trk,ie,it);
+   //    float r_ = sqrtf(x_*x_ + y_*y_);
+   //    if((it+ie*ntrks)%10==0) printf("iTrk = %i,  track (x,y,z,r)=(%.3f,%.3f,%.3f,%.3f) \n", it+ie*ntrks, x_,y_,z_,r_);
+   //  }
+   //}
+
+   int device = -1;
+   cudaGetDevice(&device);
+
+  int stream_chunk = ((int)(nevts*ntrks/num_streams));
+  int stream_remainder = ((int)((nevts*ntrks)%num_streams));
+  int stream_range;
+  if (stream_remainder == 0){ stream_range =num_streams;}
+  else{stream_range = num_streams+1;}
+  cudaStream_t streams[stream_range];
+  for (int s = 0; s<stream_range;s++){
+    //cudaStreamCreateWithFlags(&streams[s],cudaStreamNonBlocking);
+    cudaStreamCreate(&streams[s]);
+  }
+
+
+   gettimeofday(&timecheck, NULL);
+   setup_stop = (long)timecheck.tv_sec * 1000 + (long)timecheck.tv_usec / 1000;
+
+   printf("done preparing!\n");
+ 
+   printf("Number of struct MPTRK trk[] = %ld\n", nevts*nb);
+   printf("Number of struct MPTRK outtrk[] = %ld\n", nevts*nb);
+   printf("Number of struct struct MPHIT hit[] = %ld\n", nevts*nb);
+  
+   printf("Size of struct MPTRK trk[] = %ld\n", nevts*nb*sizeof(struct MPTRK));
+   printf("Size of struct MPTRK outtrk[] = %ld\n", nevts*nb*sizeof(struct MPTRK));
+   printf("Size of struct struct MPHIT hit[] = %ld\n", nlayer*nevts*nb*sizeof(struct MPHIT));
+
+   
+
+    for (int s = 0; s<num_streams;s++){
+   transferAsyncTrk(trk_dev, trk,streams[s]);
+   transferAsyncHit(hit_dev, hit,streams[s]);
+   auto wall_start = std::chrono::high_resolution_clock::now();
```

Comment:
This would actually need a synchronization point to start to clock only after the transfers are complete
```suggestion
   cudaDeviceSynchronize();
   auto wall_start = std::chrono::high_resolution_clock::now();
```
(and it should really be done better but I'd leave that to a bit later time)